aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -417,21 +456,44 @@ public Builder subscriptionsAutoReconnect( boolean subscriptionsAutoReconnect) {
          * Default value is 5 minutes. Note that this limit is for execution time - the mutation can wait in the queue for its turn to
          * be processed independent of this limit.
          * @param mutationQueueExecutionTimeout the max execution time allowed.
-         * @return
+         *
+         * @return the builder object
          */
         public Builder mutationQueueExecutionTimeout(long mutationQueueExecutionTimeout) {
             mMutationQueueExecutionTimeout = mutationQueueExecutionTimeout;
             return this;
         }
 
-        public AWSAppSyncClient build() {
-            if (mNormalizedCacheFactory == null) {
-                AppSyncSqlHelper appSyncSqlHelper = AppSyncSqlHelper.create(mContext, defaultSqlStoreName);
+        /**
+         * Flag if true will use the ClientDatabasePrefix passed in through the
+         * awsconfiguration.json or from the #clientDatabasePrefix builder.
+         *
+         * @param useClientDatabasePrefix use a prefix for the client databases.
+         * @return the builder object.
+         */
+        public Builder useClientDatabasePrefix(@NonNull boolean useClientDatabasePrefix) {
+            mUseClientDatabasePrefix = useClientDatabasePrefix;
+            return this;
+        }
 
-                //Create NormalizedCacheFactory
-                mNormalizedCacheFactory = new SqlNormalizedCacheFactory(appSyncSqlHelper);
-            }
+        /**
+         * Specify a name that uniquely identifies the AWSAppSyncClient object.
+         *
+         * @param clientDatabasePrefix prefix used for the database names.
+         * @return the builder object.
+         */
+        public Builder clientDatabasePrefix(@NonNull String clientDatabasePrefix) {
```

Comment:
same as above - consider calling this "clientPrefix" instead of "clientDatabasePrefix"

Reply:
For being explicit and transparent, we decided to use "clientDatabasePrefix". We are not introducing the concept of a client name or client identifier. And also to be consistent with the naming used by CLI, JS SDK and awsconfiguration.json.

Reply:
sounds good.